### PR TITLE
Improve documentation for list-commands

### PR DESCRIPTION
**Pull request by dominicjaeger**
Improve synopsis.
Make terms more consistent (also to existing GNU man pages)
and use more concise phrases.

Related to: #3325

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy.
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/help/kdb-list-commands.md [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
 
-`kdb list-commands`
+`kdb list-commands [OPTION]...`
 
 ## DESCRIPTION
 
 This command will list all internal kdb commands.
 The output is suitable to be processed from other
-tools as long as `-v` is not used.
+tools unless `--verbose` is specified.
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]:
```suggestion
tools unless `-v` or `--verbose` is specified.
```

File: doc/help/kdb-list-commands.md [hunk 2]
```diff
@@ -21,16 +21,16 @@ tools as long as `-v` is not used.
 - `-C`, `--color <when>`:
   Print never/auto(default)/always colored output.
 - `-v`, `--verbose`:
-  Also output some explanation for every
-  command. The command names are bold if
-  color is turned on. In the first line
-  it will add how many commands exist.
+  Print number of commands in the first line.
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]:
```suggestion
  Prints the number of commands in the first line
```

File: doc/help/kdb-list-commands.md [hunk 3]
```diff
@@ -21,16 +21,16 @@ tools as long as `-v` is not used.
...
+  Add a description to each command.
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]:
```suggestion
  and adds a description to each command.
```

File: doc/help/kdb-list-commands.md [hunk 4]
```diff
@@ -21,16 +21,16 @@ tools as long as `-v` is not used.
...
+  Print command names bold unless `--color never` is specified.
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]:
```suggestion
  Prints command names in bold unless `--color never` is specified or output is printed to non-terminals.
```

File: doc/help/kdb-list-commands.md [hunk 5]
```diff
@@ -21,16 +21,16 @@ tools as long as `-v` is not used.
...
 - `-0`, `--null`:
   Use binary 0 termination
 
 ## EXAMPLES
 
-To get a list of all available commands with their help text:
+To print a list of all internal kdb commands with their description:
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]:
```suggestion
To print a list of all internal kdb commands with their description use:
```

File: doc/help/kdb-list-commands.md [hunk 6]
```diff
@@ -2,13 +2,13 @@
 
 ## SYNOPSIS
 
-`kdb list-commands`
+`kdb list-commands [OPTION]...`
```
Comment (markus2330) [inline doc/help/kdb-list-commands.md]: This is inconsistent to the other man pages. We should either have `[OPTIONS]` everywhere or nowhere. The 3 dots at the end are misleading (there is not any other argument taken here).

File: doc/help/kdb.md [hunk 7]
```diff
@@ -22,6 +22,7 @@ In this manual we give an overview of the tool suite
 `kdb`. It is part of Elektra’s tools. The tool suite `kdb` consists
 of individual commands. Most commands are independent and some commands
 are sharing an executable. Some commands are written as external scripts.
+Subcommands have to be the first after `kdb`
```
Comment (markus2330) [inline doc/help/kdb.md]: Dot at end of sentence missing. The sentence is better placed after the first example (which does not show a command). Furthermore, the terminology is "commands" here, not subcommands (sorry that I used the wrong word before).

**Comment by markus2330** [comment]
@d3nwp Maybe you can also take a look here. This is a very nice improvement.

File: doc/help/kdb-list-commands.md (hunk 6 above)
Comment (Chemin1) [reply doc/help/kdb-list-commands.md]: How man pages work is clearly defined in [man(1)](http://man7.org/linux/man-pages/man1/man.1.html). It says, i.a.

> The command or function illustration is a pattern that should match all possible invocations.

What is currently in master does not do that, thus I added it.

>  [expression] ...   entire expression within [ ] is repeatable

And as far as I can tell using multiple options is possible: `kdb list-commands -C always -v -0`, thus using 3 dots looks correct to me.

My proposition adheres to the synopsis of GNU coreutils like [chown](http://man7.org/linux/man-pages/man1/chown.1.html), [chmod](http://man7.org/linux/man-pages/man1/chmod.1.html) and [ls](http://man7.org/linux/man-pages/man1/ls.1.html) and I think that we should apply this to other man pages, too.

Comment (markus2330) [reply doc/help/kdb-list-commands.md]: It is unfortunately not consistently done across different tools, even not within the man tools themselves e.g. `whereis` no dots, `apropos` lists the possible options, ....

> thus using 3 dots looks correct to me.

rsync uses `[OPTION...]`

But this is actually not relevant, I was only complaining that we should consistent **within ourselves**. And we were consistent before, and this PR was the first to add `[OPTION]...`.

So either we add it everywhere, or leave it as is (which sounds to me the better way, as it is not really an important thing).

Comment (mpranj) [reply doc/help/kdb-list-commands.md:5]: @Chemin1 could you please rebase and fix this little thing. It would be great to merge this work.

**Pull request closed without merging**